### DISCUSSION TITLE
Having trouble creating a new SQL script tab contextual to the DB or schema I'm already working with

### DISCUSSION THREAD

**Discussion by givihuda** [dbeaver/dbeaver#12266] [General]
I typically work with one server/schema at a time, but trying to open a new SQL script tab based on the context or view I'm in seems to provide a random context for the script, so that sometimes it'll think I'm not in any context, or in a certain connection's context, or a certain schema's context... I really just want it to open the new tab based on the last context I was in, or one related to the current view I'm in (like a table's Data or Diagram tab). This means I don't get the error that it doesn't know what server I mean, and then I get the suggestions of tables and columns without having to specify the schema in the script. So far none of my schemas or connections will cross over in SQL scripts.

Right now the last new script tab opened with `<localhost> Script-7` as its title, and it understood that I was connected to the localhost server and my specific schema; the next one opens as `<localhost> Script-8` with localhost but no awareness of which schema I mean even though I've only been working with the table tabs of the same last schema I've been working in all day.

How do I get it to understand I want the new tab in the current connection/schema I'm in, or the last one? Or what am I missing that I need to click to make this easier? I know there's several ways to open a new SQL script tab, but right-clicking a lot of things (like a table tab or a table in the Projects pane) doesn't give me the option to open a new SQL tab in that context, and I'm not seeing how to switch context in a way that it understands from the ways to open new script tabs.

**Comment by sknextcode** [comment]
You can set the current connection and schema to be used by a new script at the top bar.
e.g.:
![image](https://user-images.githubusercontent.com/81296681/116558329-98494e80-a8ff-11eb-8a9b-afd65829d93a.png)

Clicking on SQL > new SQL script should then open a tab in the context shown.

For better understanding the labeling of tabs:
As I understand, the naming of your script tabs is controlled by a pattern that you can simply change in the global settings > SQL-Editor > Scripts > Script title pattern.
Try changing the pattern from `<${connectionName}> ${fileName}` to something that is more useful for you.
You may want to add the schema by adding a `${schema}` (e.g. `<${connectionName}> ${schema} ${fileName} `).

In addition: Hovering with your mouse over the tab should display the schema the tab belongs to.

I hope that this information is helping you. If your Scripts will always open with a random context, I believe this is a bug, because my script tabs seem to be perfectly open in the context I'm in.

**Reply by givihuda** [reply]
Thanks, I notice now each tab will save what's in the schema drop-down above the tabs.

However, I'd like to correct your reference to the tab name settings, which I found in Preferences > Editors > SQL Editor > Scripts.

![image](https://user-images.githubusercontent.com/74785256/122964427-7dfe8080-d344-11eb-8fa9-cb2a6d301160.png)

I do notice though that if I click the SQL drop-down at the top left, that will open a new SQL tab in the current-selected connection and schema:

![image](https://user-images.githubusercontent.com/74785256/122964734-cae25700-d344-11eb-8683-54fb4f565b9f.png)

So maybe that's the best solution. It's certainly better than scrolling up and right- clicking the schema name again.

What would be easier is a keyboard shortcut to do the same thing... which I'm not finding yet. Cmd+T doesn't do it.
